### PR TITLE
feat: space page

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: https://github.com/lightdash/lightdash/issues/3860

### Description:

- [x] All Spaces should live on a new page /spaces
- [x]  You should be able to create a Space from here
- [x]  Add the page 'All Spaces' to the “Browse” navbar
- [x]  On individual Space pages, change the breadcrumbs from Home > [Space Name] to All Spaces > [Space Name] and make the breadcrumb link back to the new Spaces page instead.

https://user-images.githubusercontent.com/962095/214005284-cd851211-e2fc-408b-a40c-fe9a06f37b2d.mp4

### CODE REVIEW

**Comment by IrakliJani** [comment]
@PriPatel UI bits here: 
1. space icon has classic folder icon. let me know if you want something different. or you want to group Space menu item with spaces list below
CleanShot 2023-01-23 at 13 59 57@2x

2. I did not extract "create space" button like we have in other "list pages" like dashboards. I believe this spaces grid will be a list component (or a list + grid option for the user) and require structural changes anyways.
CleanShot 2023-01-23 at 13 59 48@2x

File: packages/frontend/src/App.tsx [hunk 1]
```diff
@@ -262,6 +263,16 @@ const App = () => (
                                                                         <Space />
                                                                     </TrackPage>
                                                                 </Route>
+                                                                <Route path="/projects/:projectUuid/spaces">
+                                                                    <NavBar />
+                                                                    <TrackPage
+                                                                        name={
+                                                                            PageName.SPACE
```
Comment (ZeRego) [inline packages/frontend/src/App.tsx:270]: What do you think if we changed this to SPACES and the SPACES in line 260 to SPACE?
This is going to screw up a bit the analytics but makes more sense.

Comment (IrakliJani) [reply packages/frontend/src/App.tsx:270]: yeah, makes sense

**Comment by ZeRego** [comment]
Found odd `all spaces` and the individual spaces menu items have the same icon. There isn't a clear separation between those 2. Perhaps moving the `all spaces` above all dashboards and all charts would make more sense? 🤔 
 
Screenshot 2023-01-23 at 11 27 30

I assume we don't want to change the homepage in this PR to have `show all` like the dashboard table. 
Screenshot 2023-01-23 at 11 28 40

**Review by ZeRego — APPROVED** [review]

**Comment by PriPatel** [comment]
@IrakliJani Thanks!
1. Lets use the `Projects` icon and place the item after `all dashboard` and `all charts`
2. On your comment here:
> I did not extract "create space" button like we have in other "list pages" like dashboards. I believe this spaces grid will be a list component (or a list + grid option for the user) and require structural changes anyways.

Makes sense to me - many structural changes coming!

**Pull request merged**